src/jinja2/parser.py

```diff
@@ -254,7 +254,12 @@ def parse_autoescape(self):
     def parse_block(self):
         node = nodes.Block(lineno=next(self.stream).lineno)
         node.name = self.stream.expect("name").value
-        node.scoped = self.stream.skip_if("name:scoped")
+
+        node.scoped = False
+        node.required = False
+        while self.stream.current.type == "name":
+            setattr(node, self.stream.current.value, True)
+            next(self.stream)
```

Comment:
I think this block might allow invalid syntax, such as a typo like "scope" instead of "scoped", or other words entirely. It would assign an unused variable, but instead it should raise an error. Would you confirm whether this happens?

Reply:
Yup you were right. I added a `test_any` before the assignment and raised an error if it fails to fix this. 

This made me realize that duplicate assignments are also allowed (e.g. `{% block body scoped scoped %}` will still go through. Is this a problem we should be concerned about?

Reply:
> This made me realize that duplicate assignments are also allowed (e.g. `{% block body scoped scoped %}` will still go through. Is this a problem we should be concerned about?

@davidism Just had an idea to address the concern above:
```python
node.scoped = self.stream.skip_if("name:scoped")
node.required = self.stream.skip_if("name:required")
        
node.scoped = self.stream.skip_if("name:scoped") if not node.scoped else node.scoped
node.required = self.stream.skip_if("name:required") if not node.required else node.required
```
This implementation throws an error for invalid syntax as well as duplicated attributes. Would this be a better approach?

Reply:
Maybe we should to add a "one of" function to the lexer?

Reply:
Ooh that's a good idea! So it'll be like `test_any` but for `skip_if`'s

Reply:
Hmm once I started implementing it, I ran into the same issue of having to know which attribute to set, and setting the attribute to False if not present. This is given that `one_of` has `next_if`'s functionality but over an iterable:
```python
node.scoped = False
node.required = False

token = self.stream.one_of("name:scoped", "name:required")
if token is not None:
    setattr(node, token.value, True)
token = self.stream.one_of("name:scoped", "name:required")
if token is not None:
    setattr(node, token.value, True)
``` 
It doesn't seem much cleaner unless I interpretted `one_of` differently.



Another idea was to add an `look_and_skip_if` to the lexer which operates like `skip_if` but checking the next token as well if the current does not match. 
```python
node.scoped = self.stream.look_and_skip_if("name:scoped")
node.required = self.stream.look_and_skip_if("name:required")
```
I'm leaning towards the second idea, but would love to get your input first.

Reply:
`look_and_skip_if` looks good to me.